lisa/util/shell.py

```diff
@@ -291,24 +292,43 @@ def spawn(
         self.initialize()
         assert self._inner_shell
 
-        try:
-            process: spur.ssh.SshProcess = _spawn_ssh_process(
-                self._inner_shell,
-                command=command,
-                update_env=update_env,
-                store_pid=store_pid,
-                cwd=cwd,
-                stdout=stdout,
-                stderr=stderr,
-                encoding=encoding,
-                use_pty=use_pty,
-                allow_error=allow_error,
-            )
-        except FunctionTimedOut:
-            raise LisaException(
-                f"The remote node is timeout on execute {command}. "
-                f"It may be caused by paramiko/spur not support the shell of node."
-            )
+        MAX_TRIES = 3
```

Comment:
I'm not preferring to try from this layer. When the command is time out, it's hard to know it's executed or not. The blinding retrying on the lower layer may make the command execute multiple times. So the retry should be write on higher layer, which the result is fully understood. Either it's ok to run multiple times, or the reason of timeout is clear.